rdkit/Chem/QED.py

```diff
@@ -0,0 +1,317 @@
+      [2.948620388, 160.4605972, 3.615294657, 4.435986202, 0.290141953, 1.300669958, 148.7763046],
+      [1.618662227, 1010.051101, 0.985094388, 0.000000001, 0.713820843, 0.920922555, 258.1632616],
+      [1.876861559, 125.2232657, 62.90773554, 87.83366614, 12.01999824, 28.51324732, 104.5686167],
+      [0.010000000, 272.4121427, 2.558379970, 1.565547684, 1.271567166, 2.758063707, 105.4420403],
+      [3.217788970, 957.7374108, 2.274627939, 0.000000001, 1.317690384, 0.375760881, 312.3372610],
+      [0.010000000, 1199.094025, -0.09002883, 0.000000001, 0.185904477, 0.875193782, 417.7253140]    ]
+
+
+def ads(x, a, b, c, d, e, f, dmax): #pylint: disable=R0913
+  """ ADS function """
+  return ((a + (b / (1 + exp(-1 * (x - c + d / 2) / e)) * (1 - 1 / (1 + exp(-1 * (x - c - d / 2) / f))))) / dmax)
+
+
+def properties(mol):
+  """
+  Calculates the properties that are required to calculate the QED descriptor.
+  """
+  matches = []
+  if (mol is None):
+    raise TypeError('You need to provide a mol argument.')
+  x = [0] * 8
+  x[0] = rdmd._CalcMolWt(mol)                        # MW 
+  x[1] = Crippen.MolLogP(mol)                        # ALOGP
+  for hbaPattern in Acceptors:                            # HBA
+    if (mol.HasSubstructMatch(hbaPattern)):
+      matches = mol.GetSubstructMatches(hbaPattern)
+      x[2] += len(matches)
+  x[3] = Lipinski.NumHDonors(mol)               # HBD
+  x[4] = MolSurf.TPSA(mol)                        # PSA
+  x[5] = Lipinski.NumRotatableBonds(mol)         # ROTB
+  x[6] = Chem.GetSSSR(Chem.DeleteSubstructs(deepcopy(mol), AliphaticRings))  # AROM
+  for alert in StructuralAlerts:                        # ALERTS
+    if (mol.HasSubstructMatch(alert)): x[7] += 1
+  return x
+
+
+def qed(m=None,w=(0.66, 0.46, 0.05, 0.61, 0.06, 0.65, 0.48, 0.95),
+        p=None):
+  """ Calculate the weighted sum of ADS mapped properties
+
+  some examples from the QED paper, reference values from Peter G's original implementation
+  >>> m = Chem.MolFromSmiles('N=C(CCSCc1csc(N=C(N)N)n1)NS(N)(=O)=O')
+  >>> qed(m)
+  0.241...
+  >>> m = Chem.MolFromSmiles('CNC(=NCCSCc1nc[nH]c1C)NC#N')
+  >>> qed(m)
+  0.217...
+  >>> m = Chem.MolFromSmiles('CCCCCNC(=N)NN=Cc1c[nH]c2ccc(CO)cc12')
+  >>> qed(m)
+  0.212...
+  """
+  if p is None:
+      p = properties(m)
+  d = [0.00] * 8
+  for i in range(0, 8):
+    d[i] = ads(p[i], pads[i][0], pads[i][1], pads[i][2], pads[i][3], pads[i][4], pads[i][5], pads[i][6])
+  t = 0.0
+  for i in range(0, 8):
+    t += w[i] * log(d[i])
+  return (exp(t / sum(w)))
+  
+  
+def weights_max(mol):
+  """
+  Calculates the QED descriptor using maximal descriptor weights.
+  """
+  props = properties(mol)
+  return qed(mol,w=[0.50, 0.25, 0.00, 0.50, 0.00, 0.50, 0.25, 1.00])
+
+
+def weights_mean(mol):
+  """
+  Calculates the QED descriptor using average descriptor weights.
+  """
+  props = properties(mol)
+  return qed(mol,w=[0.66, 0.46, 0.05, 0.61, 0.06, 0.65, 0.48, 0.95])
```

Comment:
These are the same weights as the default weights in the function signature. (I assume). 
```
return qed(mol)
```